### PR TITLE
Connect real data to Overview

### PR DESCRIPTION
**Pull request by viktor-yakubiv**
@Joozty I would like to get your early feedback about connecting store. I am afraid I miss calling events somewhere and do not handle async methods. Could you please point me where to look at and what is the best way to fetch statistics data if it is needed only once at this moment?

On top of #81. Please, check the last commit only.

---

**UPD:** This PR should have been a draft but I have mistaken.

### CODE REVIEW

**Comment by Joozty** [comment]
I would use `autorun` for it. Something similar as it is done [here](https://github.com/oacore/dashboard/blob/master/store/store.js#L46).


```
class Statistics {
  @observable statistic = {}

  constructor(rootStore) {
    this.rootStore = rootStore

    this.ononDataProviderChange()
  }
  
  retrieveStatistics = async () => {
    try {
      const [data] = await apiRequest(
        `/data-providers/${this.rootStore.dataProvider}/statistics`,
        'GET',
        {},
        true
      )
	  this.statistics = data 
    } catch (e) {
      if (e.statusCode !== 404) throw e
    }
  }
  
  onDataProviderChange = () =>
    autorun(() => {
		this.retrieveStatistics()
    })
}
```


Or you can use retrieveStatistics method in componentDidMount.

**Comment by Joozty** [comment]
When you use autorun. Request gets fired even when you are not on that page. I want to fix it in the future. But for now, we need to live with it.

**Comment by viktor-yakubiv** [comment]
> When you use autorun. Request gets fired even when you are not on that page. I want to fix it in the future. But for now, we need to live with it.

It's fine. With some endpoints we will need to do it anyway.

Thank you for your advice.

File: api/index.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import axios from 'axios'
 import { NetworkError } from './errors'
 
 const CORE_API = 'https://api.core.ac.uk/internal'
-const CORE_API_DEV = 'https://api.dev.core.ac.uk/internal'
```
Comment (Joozty) [inline api/index.js]: Not a good idea. A cookie will not be sent.

File: store/statistics.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class Statistics {
+  constructor(rootStore) {
+    this.rootStore = rootStore
+  }
+
+  retrieveStatistics = async dataProviderId => {
+    try {
+      const { data } = await apiRequest(
+        `/data-providers/${dataProviderId}/statistics`,
```
Comment (Joozty) [inline store/statistics.js]: I was using it like this in my actions:

`/data-providers/${this.rootStore.dataProviderId}/statistics`

File: store/statistics.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      )
+
+      return data || {}
+    } catch (e) {
+      if (e.statusCode !== 404) throw e
+      return {}
+    }
+  }
+
+  onDataProviderChange = async dataProviderId => {
```
Comment (Joozty) [inline store/statistics.js]: would be nice to mark this as an `@action`

File: api/index.js (hunk 1 above)
Comment (viktor-yakubiv) [reply api/index.js]: Yeah, I know 😔

File: store/statistics.js (hunk 2 above)
Comment (viktor-yakubiv) [reply store/statistics.js]: Please, check the current solution. Does it satisfy you?

I try to make functions as pure as possible. Maybe here it's not the best what I could do. Let me know.

File: store/statistics.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      return {}
+    }
+  }
+
+  @action
+  onDataProviderChange = async (
+    dataProviderId = this.rootStore.dataProvider
+  ) => {
+    const statistics = await this.retrieveStatistics(dataProviderId)
+    Object.assign(this, statistics)
```
Comment (viktor-yakubiv) [inline store/statistics.js]: @Joozty I saw you used similar approach. Please, check in line 14 how I solved the issue with this. Line from the MobX's documentation:
> Only plain objects will be made observable. For non-plain objects it is considered the responsibility of the constructor to initialize the observable properties. Either use the `@observable` annotation or the `extendObservable` function.

**Pull request merged**